client/src/op/message.rs

```diff
@@ -656,6 +655,26 @@ impl Message {
                 key_tag,
                 // can probably get rid of this clone if the owndership is correct
                 signer.signer_name().clone(),
+                Vec::new(),
+            )
+        ));
+        let signature: Vec<u8> = try!(signer.sign_message(self, &sig0));
+        sig0.set_rdata(
+            RData::SIG(SIG::new(
```

Comment:
Sure. Did consider this myself, but as all the `RData::*` types seem to be immutable, this would be a little bit inconsequential.
But you are right, the current code is bug-prone, so adding a `set_sig` function may be the better choice.